src/Maths/math.vector.ts

```diff
@@ -47,8 +51,11 @@ export class Vector2 {
      * @returns the Vector2 hash code as a number
      */
     public getHashCode(): number {
-        let hash = this.x | 0;
-        hash = (hash * 397) ^ (this.y | 0);
+        const x = _ExtractAsInt(this.x);

```

Comment:
Is this not a fix for a bug on the user side?

`x` and `y` are declared as numbers so they should not hold string data. If they are, it means the user put wrong (invalid) values there. I don't know at which frequency `getHashCode` can be called, but calling `_ExtractAsInt` two times is probably taking most of the time of the whole function time.

Reply:
nope the problem was that this.x | 0 will convert to an int. So losing everything after the comma. 

The function is not meant to be fast anyway

Reply:
https://forum.babylonjs.com/t/gethashcode-of-vector3-class-does-not-work-on-some-x-y-z-values/26519/5

Reply:
> So losing everything after the comma.

You will also loose everything after the comma with the current PR. Maybe you want ot use `parseFloat` instead of `parseInt`?

Reply:
> You will also loose everything after the comma with the current PR 

How's that? The string get stripped of the "." and thus it becomes a long int.

> The fix could not be to simply remove | 0

The hash needs to be an int

Reply:
Yes, too many holidays are not good for the brain, sorry for the noise :)

Reply:
haha no worries! Enjoy your holidays!